### PR TITLE
luci-app-usteer: Added tooltip in status page

### PR DESCRIPTION
**Pull request by Ramon00**
Added tooltip while hovering over connected client showing mac, IP(s), hostname in status page

### CODE REVIEW

File: applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js [hunk 1]
```diff
@@ -91,25 +91,55 @@ function collectWlanAPInfoEntries(connectioninfo_table_entries, wlanAPInfos) {
 	}
 };
 
+function tootltip(mac, IP, hostname) {
+	var body= E([]);
+	var i=0;
+	body.appendChild(E('div', mac));
+	if (typeof IP !== 'undefined') {
+		while ((typeof IP['ipaddrs'][i] !== 'undefined') && (String(IP['ipaddrs'][i]).length > 0)) {
```
Comment (systemcrash) [inline applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js:99]: Please revise these `while` loops to `forEach` loops. This makes them less ambiguous and avoids variable `i` declaration and assignment errors. You also avoid indices.

File: applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js [hunk 2]
```diff
@@ -91,25 +91,55 @@ function collectWlanAPInfoEntries(connectioninfo_table_entries, wlanAPInfos) {
...
+			body.appendChild(E('div', IP['ipaddrs'][i]));
+			i++;
+		}
+		
+		i=0;
+		while ((typeof IP['ip6addrs'][i] !== 'undefined') && (String(IP['ip6addrs'][i]).length > 0)) {
```
Comment (systemcrash) [inline applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js:105]: using `&& String...` might behave unexpectedly: what is the truth test here? That you coerced something to a string? Or that length is > 0? Perhaps skip String()

File: applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js [hunk 3]
```diff
@@ -91,25 +91,55 @@ function collectWlanAPInfoEntries(connectioninfo_table_entries, wlanAPInfos) {
...
 
 function collectWlanAPInfos(compactconnectioninfo_table_entries, wlanAPInfos) {
 	for (var wlan in wlanAPInfos) {
-		var hostl = '';
+		var hostl = E([]);
 		for (var mac in Clients) {
 			if (typeof Clients[mac] !== 'undefined')
 				if (typeof Clients[mac][wlan] !== 'undefined')
-					if (String(Clients[mac][wlan]['connected']).valueOf() == String('true').valueOf()) {
+					if (String(Clients[mac][wlan]['connected']).valueOf() === String('true').valueOf()) {
```
Comment (systemcrash) [inline applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js:122]: Must one coerce everything to strings here? Doesn't the following work identically?:

`Clients[mac][wlan]['connected'] === 'true'`

**Review by systemcrash — CHANGES_REQUESTED** [review]

File: applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js (hunk 1 above)
Comment (Ramon00) [reply applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js:99]: I agree it is a bit messy, I made it into `for (var IPaddr in IP['ipaddrs']) body.appendChild(E('div', IP['ipaddrs'][IPaddr]));` now. So much cleaner. Thx

File: applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js (hunk 2 above)
Comment (Ramon00) [reply applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js:105]: I tested for length >0 as I encountered a empty string as only entry in the past. With the change above this is now not needed anymore.

File: applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js (hunk 3 above)
Comment (Ramon00) [reply applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js:122]: Somehow that does not work. But `if (String(Clients[mac][wlan]['connected']).valueOf() === 'true') ` that does work. So I will leave it like this.

**Comment by systemcrash** [comment]
Do you think it's safe to do this with the mac and not null etc check it like IPs and hostname ? 
`	body.appendChild(E('div', mac));`

**Comment by Ramon00** [comment]
> Do you think it's safe to do this with the mac and not null etc check it like IPs and hostname ?

@systemcrash yes that is safe because the mac is _the_ client identifier used. So there is always a mac, there may be one or more IP addresses and there may be a host name. On the main router mosty everything is available, on the AP it may only be a mac address.

**Comment by Ramon00** [comment]
Oh that reminds me to 'escape' the hostname. Let me add that

**Comment by systemcrash** [comment]
OK. Shall we leave this to percolate for a diem and merge?

**Review by systemcrash — APPROVED** [review]

**Comment by Ramon00** [comment]
@systemcrash sounds like a plan!

**Pull request merged**